crates/typst-layout/src/inline/line.rs

```diff
@@ -559,8 +561,9 @@ pub fn commit(
         add_par_line_marker(&mut output, marker, engine, locator, top);
     }
 
+    frames.sort_by_key(|(_, _, idx)| *idx);
```

Comment:
can it happen that the sorting is something other than either a no-op or a reversal?